### ISSUE TITLE
Parsing error at Alexa in harpyNest - dev branch

### ISSUE THREAD

**Issue by Chunim** [Innoxia/liliths-throne-public#1061]
There is a parsing error at the entrance to Alexa's nest. 
this is because the guestNode have not been updated in the xml file.

com.lilithsthrone.game.dialogue.places.dominion.harpyNests.HarpyNestAlexa;
49 `Quest.QUEST_MAIN_1_E_REPORT_TO_ALEXA` -> `Quest.MAIN_1_E_REPORT_TO_ALEXA`

Another error is when the content for the DialogueNode ALEXAS_NEST_SCARLETT is called in.
The node call in the file `ALEXAS_NEST` when is should call in `ALEXAS_NEST_SCARLETT`.
This is done at the row 103.

**Comment by arpie-steele** [comment]
In res/txt/places/dominion/harpyNests/alexa.xml (0.3.0.7) at lines 30 and 57 the Java-like QuestLine.MAIN argument must be changed to QUEST_LINE_MAIN for it to be meaningful to the javax.script.ScriptEngine instance initialized by the method initScriptEngine (lines 6053-6055 of game/dialogue/utils/UtilText.java).

Upon review of the above, they may be confused since the error log message states:

Version: 0.3.0.7
Java: 1.8.0_201
Conditional parsing error: pc.getQuest(QuestLine.MAIN) == QUEST_MAIN_1_E_REPORT_TO_ALEXA
ReferenceError: "QuestLine" is not defined in <eval> at line number 1

**Comment by arpie-steele** [comment]
I have reconfirmed this fix on a fresh build of the post 0.3.0.7 dev branch.

**Step One -- create new character and finish prologue -- then save.**

- New Game
- Agree
- Continue
- Start
- Continue
- Continue
- Continue
- To the Stage
- Select Job
- Continue
- Continue
- Skip prologue
- Entrance hall
- Exit
- [Gear]
- Save/Load
- [Save as Preteleport]
- [Quit]

**Step two -- move or teleport to Alexa's nest location**

- cp data/saves/Preteleport.xml data/saves/Postteleport.xml
- edit data/saves/Postteleport.xml
- In the new file Change:
  - game.playerCharacter.character.locationInformation.worldLocation.value from DOMINION to HARPY_NEST
  - game.playerCharacter.character.locationInformation.location.x from 8 to 2
  - game.playerCharacter.character.locationInformation.location.y from 8 to 7
- [Save/Load]
- [load Postteleport]
- [Verify error message: (Error in conditional parsing!)]
- [Quit]

There is also a related message in data/error.log which is confusing because QuestLine.MAIN is correct syntax for Java.

**Step three: Patch res/txt/places/dominion/harpyNests/alexa.xml and rebuild and relaunch**

- Basically, wherever it used to say {{QuestLine.MAIN}} replace with {{QUEST_LINE_MAIN}}

```diff
diff --git a/res/txt/places/dominion/harpyNests/alexa.xml b/res/txt/places/dominion/harpyNests/alexa.xml
index e8fd3442..4f5bb5ab 100644
--- a/res/txt/places/dominion/harpyNests/alexa.xml
+++ b/res/txt/places/dominion/harpyNests/alexa.xml
@@ -27,7 +27,7 @@
                Meeting with Alexa by appointment only.</i>
        </p>
        <p>
-               #IF pc.getQuest(QuestLine.MAIN) == QUEST_MAIN_1_E_REPORT_TO_ALEXA
+               #IF pc.getQuest(QUEST_LINE_MAIN) == QUEST_MAIN_1_E_REPORT_TO_ALEXA
                #THEN
                        Stepping back, you wonder if you should call over one of the nearby harpies. After all, you do have business with Alexa, and you're sure that she'd want to hear about Scarlett as soon as possible.
                #ELSE
@@ -54,7 +54,7 @@
                Meeting with Alexa by appointment only.</i>
        </p>
        <p>
-               #IF pc.getQuest(QuestLine.MAIN) == QUEST_MAIN_1_E_REPORT_TO_ALEXA
+               #IF pc.getQuest(QUEST_LINE_MAIN) == QUEST_MAIN_1_E_REPORT_TO_ALEXA
                #THEN
                        You do have business with Alexa, and you're sure that she'd want to hear about Scarlett as soon as possible, but you'll have to come back after the ongoing storm has passed.
                #ELSE
```
**Step Four: Verify parsing problem gone**

**Comment by arpie-steele** [comment]
With the merge, this tests as fixed in builds of dev branch.

**Comment by Innoxia** [comment]
Fixed as arpie-steele says in PR#1069
